presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorDistributedQueries.java

```diff
@@ -145,4 +150,86 @@ public void testCreateBucketedTable()
 
         assertUpdate("DROP TABLE orders_bucketed");
     }
+
+    @Test
+    public void testTablesSystemTable()
+    {
+        assertUpdate("" +
+                "CREATE TABLE system_tables_none(t1 timestamp, o1 varchar, o2 double, b1 bigint, b2 bigint) " +
+                "WITH (temporal_column = 't1')");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_ordered(t3 timestamp, o3 varchar, o4 double, b3 bigint, b4 bigint) " +
+                "WITH (temporal_column = 't3', ordering = ARRAY['o3', 'o4'])");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_temporal_bucketed(t5 timestamp, o5 varchar, o6 double, b5 bigint, b6 bigint) " +
+                "WITH (temporal_column = 't5', bucket_count = 40, bucketed_on = ARRAY ['b5', 'b6'])");
+        assertUpdate("" +
+                "CREATE TABLE system_tables_temporal_ordered_bucketed(t7 timestamp, o7 varchar, o8 double, b7 bigint, b8 bigint) " +
+                "WITH (temporal_column = 't7', ordering = ARRAY['o7', 'o8'], distribution_name = 'test_distribution', bucket_count = 50, bucketed_on = ARRAY ['b7', 'b8'])");
+
+        MaterializedResult actualResults = computeActual("SELECT * FROM system.tables");
+        assertEquals(actualResults.getTypes(), ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, new ArrayType(VARCHAR), VARCHAR, BIGINT, new ArrayType(VARCHAR)));
```

Comment:
It will make this test easier to read if we list the column names here (so the reader doesn't have to go look for them elsewhere):
```java
assertEquals(actualResults.getTypes(), ImmutableList.builder()
        .add(VARCHAR) // table_schema
        .add(VARCHAR) // table_name
        .add(VARCHAR) // temporal_column
        .add(new ArrayType(VARCHAR)) // ordering_columns
        .add(VARCHAR) // distribution_name
        .add(BIGINT) // bucket_count
        .add(new ArrayType(VARCHAR)) // bucket_columns
        .build());